OctopusDSC/DSCResources/cTentacleAgent/cTentacleAgent.psm1

```diff
@@ -1002,9 +1129,17 @@ function Add-TentacleToWorkerPool
             "register-worker",
             "--instance", $name,
             "--server", $octopusServerUrl,
-            "--force"
+            "--force",
+            "--comms-style", $commsStyle
         )
 
+        # Get tentacle version to determine spaces compatiblity
+        if ([System.Version](Get-TentacleVersion) -ge [System.Version]"4.0.0")
```

Comment:
This code is in a bunch of different locations. Can you pull it into a separate function?
Check out how the server does it.

Reply:
Function removed entirely.